californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -478,14 +478,13 @@ private boolean isResponseRelatedToRequest(final Exchange exchange, final Correl
 			// no correlation information available for request, thus any
 			// additional correlation information available in the response is ignored
 			return true;
-		} else if (exchange.getCorrelationContext() instanceof DtlsCorrelationContext) {
+		} else if (exchange.getCorrelationContext().get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
```

Comment:
Yes, it is true about the first point of you.
We can not store the CorrelationContext Object (in MongoDB or Redis), we only will store the information about the keys (Therfore the getKeys() method in CorrelationContext). 
We can restore it to a MapBasedCorrelationContext, if its a DtlsCorrelationContext the KEY_SESSION_ID is available.
Besides, I am of the opinion, **instanceOf** is not best practice.
I do not quite understand the second point. If there may be no other CorrelationContext as DtlsCorrelationContext should we then not only use this?